Modules/Loadable/Segmentations/Widgets/qMRMLSegmentEditorWidget.cxx

```diff
@@ -1141,6 +1147,13 @@ void qMRMLSegmentEditorWidget::updateEffectList()
     QLayoutItem *child;
     while ((child = d->EffectsGroupBox->layout()->takeAt(0)) != 0);
     }
+  QList<QAbstractButton*> undoRedobuttons = d->UndoRedoButtonGroup.buttons();
+  foreach (QAbstractButton* button, undoRedobuttons)
+    {
+    button->hide();
+    QLayoutItem *child;
+    while ((child = d->UndoRedoGroupBox->layout()->takeAt(0)) != 0);
```

Comment:
Are you sure you need to remove the widgets from the layout? I think if you add them to the desired layout position then if the position is different then they are removed automatically (but if the position remains the same then nothing is changed and we can avoid potential flickering).

Reply:
I did this just to make it similar logic as the effect frame in case other buttons made it into this new frame. I think you’re right that adding all the widgets in the grid layout again in a different position when redefining effect column count that it will remove the conflict of a widget already present in the specified cell.

Reply:
The effect buttons initialization is not perfect. On some computers (maybe slower ones, maybe better visible in debug mode and/or when many effects are registered) there is some flickering when the segment editor widget displayed first. Maybe the buttons are momentarily appear over the screen before they got moved into the layout. I would not delay this pull request with investigation of this small issue, just mention it because it might not be ideal to copy how the effect button layout is created/updated.